src/test/java/org/jenkinsci/plugins/pipeline/utility/steps/fs/TeeStepTest.java

```diff
@@ -76,6 +77,49 @@ public void smokes() throws Exception {
         });
     }
 
+    @Test
+    @Issue({"JENKINS-54346", "JENKINS-55505"})
+    public void closed() throws Exception {
+        rr.then(r -> {
+            r.createSlave("remote", null, null);
+            WorkflowJob p = r.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition(
+                    "node('remote') {\n" +
+                            "  tee('x.log') {\n" +
+                            "    echo 'first message'\n" +
+                            "  }\n" +
+                            "  if (isUnix()) {sh 'rm x.log'} else {bat 'del x.log'}\n" +
+                            "  writeFile file: 'x.log', text: 'second message'\n" +
```

Comment:
Huh? What is this testing? You are deleting a file, rewriting it with some new contents, reading those same contents back, and asserting you read what you just wrote?

Reply:
It tests that the file gets closed properly so it can be deleted, rewritten and the new content read back. This test the exact scenario of the JENKINS-54346 bug where the file wasn't closed after the tee step.

Reply:
To reproduce that I think it would suffice to just have the `del` on Windows, right?